### PR TITLE
Feature/smar 41

### PR DESCRIPTION
**Pull request by RaznoOleg**
Description:
• Added endpoint for forgot password - we enter an email, and if a user with such an email exists in the system, a link is sent to his mail, clicking on which will open a form for changing the password;
• Added endpoint for reset password;

P.S.: branch created not from develop, but from feat/register+email branch

### General
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] All links in on ENV files
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Backend
- [x] swagger for each endpoint
- [x] less requests to db, all data should be taken with one query
- [x] public in method use only is use function externally
- [x] use ConfigService instead of process.env

#### Successful post forgot password:
image

#### The link that the user received by the entered email:
image

#### Unsuccessful post forgot password:
image

#### Successful patch reset password:
image

#### Successful login with a new password:
image

#### Unsuccessful patch reset password if we entered the wrong password:
image

#### Unsuccessful patch reset password if invalid token:
image

### CODE REVIEW

File: src/modules/auth/auth.service.ts [hunk 1]
```diff
@@ -49,9 +111,146 @@ export default class AuthService {
...
+    );
+
+    res.cookie('accessToken', accessToken, this.accessTokenCookieOptions);
+    res.redirect(this.configService.get('CLIENT_URL'));
+  }
+
+  private async getGoogleOauthTokens(code: {
+    code: string;
+  }): Promise<Record<string, string>> {
+    const url = 'https://oauth2.googleapis.com/token';
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: move to constants or env

File: src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -49,9 +111,146 @@ export default class AuthService {
...
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  private async getGoogleUser({
+    id_token,
+    access_token,
+  }): Promise<GoogleDoctorResult> {
+    try {
+      const res = await axios.get(
+        `https://www.googleapis.com/oauth2/v1/userinfo?alt=json&access_token=${access_token}`,
```
Comment (lyudad) [inline src/modules/auth/auth.service.ts]: move to constants or env

File: src/modules/doctor/dto/update-google-doctor-dto.ts [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  IsISO31661Alpha2,
+  IsNumber,
+} from 'class-validator';
+import {
+  CITY_REGEX,
+  ADDRESS_REGEX,
+  DATE_REGEX,
+  TIME_ZONE_REGEX,
+} from '../../../shared/consts';
+import { Role, Gender } from '../../../shared/enums';
```
Comment (lyudad) [inline src/modules/doctor/dto/update-google-doctor-dto.ts]: change to absolute path

File: src/modules/auth/auth.service.ts (hunk 1 above)
Comment (RaznoOleg) [reply src/modules/auth/auth.service.ts]: fixed

File: src/modules/auth/auth.service.ts (hunk 2 above)
Comment (RaznoOleg) [reply src/modules/auth/auth.service.ts]: fixed

File: src/modules/doctor/dto/update-google-doctor-dto.ts (hunk 3 above)
Comment (RaznoOleg) [reply src/modules/doctor/dto/update-google-doctor-dto.ts]: fixed

**Pull request merged**